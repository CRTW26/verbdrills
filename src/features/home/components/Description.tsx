import Button from 'shared/components/Button'

interface Props {
  text: string
}

const Description: React.FC<Props> = ({ text }) => {
  return (
    <div className="description" data-testid="description">
      <h3>{text}</h3>
      <Button text="PRACTICE" />
    </div>
  )
}

export default Description
