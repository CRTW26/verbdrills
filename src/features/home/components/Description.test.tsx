import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Description from './Description'

describe('<Description />', () => {
  const text =
    'Nail verb conjugation to help you on your journey to being a language master'

  test('should render', () => {
    const { asFragment } = render(<Description text={text} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should have correct heading', () => {
    const { getByTestId } = render(<Description text={text} />)

    expect(getByTestId('description')).toHaveTextContent(`${text}`)
  })
})
