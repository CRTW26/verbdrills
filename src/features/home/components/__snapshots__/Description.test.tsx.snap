// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> should render 1`] = `
<DocumentFragment>
  <div
    class="description"
    data-testid="description"
  >
    <h3>
      Nail verb conjugation to help you on your journey to being a language master
    </h3>
    <button
      class="jsx-2467018731 "
      data-testid="btn--train"
    >
      PRACTICE
    </button>
  </div>
</DocumentFragment>
`;
